### PR TITLE
chore: update contributing guide with new process for unconventional dependencies

### PR DESCRIPTION
**Pull request by kaizencc**
----

### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: CONTRIBUTING.md
Comment (Naumel) [inline CONTRIBUTING.md:399]: [nit] Do we have a public list that we could point to?

File: CONTRIBUTING.md
Comment (Naumel) [inline CONTRIBUTING.md:407]: [nit] This is not a different option, it's an increment of the step above -- right?
Otherwise "in addition" reads strangely in the paragraph

File: CONTRIBUTING.md
Comment (Naumel) [inline CONTRIBUTING.md:412]:
```suggestion
[construct](https://github.com/aws/aws-cdk/blob/main/packages/@aws-cdk/aws-eks/README.md?plain=1#L685-L692).
```

File: CONTRIBUTING.md
Comment (Naumel) [inline CONTRIBUTING.md:414]:
```suggestion
**If your PR introduces runtime dependencies in lambda layers, make sure to call it out in the description
```

**Review by Naumel — DISMISSED** [review]
Couple of nit-picks, I'd like to see the typos fixed and I'll switch to `approve` 😊

**Review by madeline-k — COMMENTED** [review]
I added my feedback by just pushing to the branch. I figured that was easier than commenting in this case. @kaizencc, let me know what you think!

File: CONTRIBUTING.md
Comment (madeline-k) [reply CONTRIBUTING.md]: We don't have one, and it would be too annoying to maintain :/ I do agree, it would be nice though.

File: CONTRIBUTING.md
Comment (madeline-k) [reply CONTRIBUTING.md]: +1, I think this is addressed in my review.

**Review by kaizencc — COMMENTED** [review]
I think the code looks good now, but I cannot approve as I am the original author of the PR. I have added an additional commit that changes locations where we reference `Unconventional Dependencies` to instead reference `Construct Runtime Dependencies` that need a second pair of eyes as well.

**Review by madeline-k — APPROVED** [review]

**Pull request merged**
